src/main/java/com/github/quiltservertools/ledger/mixin/FillCommandMixin.java

```diff
@@ -0,0 +1,34 @@
+package com.github.quiltservertools.ledger.mixin;
+
+import com.github.quiltservertools.ledger.callbacks.BlockChangeCallback;
+import com.github.quiltservertools.ledger.utility.Sources;
+import net.minecraft.block.BlockState;
+import net.minecraft.command.argument.BlockStateArgument;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.server.command.FillCommand;
+import net.minecraft.server.command.ServerCommandSource;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.math.BlockPos;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+@Mixin(FillCommand.class)
+public class FillCommandMixin {
+    @Redirect(method = "execute", at = @At(value = "INVOKE", target = "Lnet/minecraft/command/argument/BlockStateArgument;setBlockState(Lnet/minecraft/server/world/ServerWorld;Lnet/minecraft/util/math/BlockPos;I)Z"))
```

Comment:
Should be a ModifyArgs mixin without redirecting that call

Reply:
Unlike probably most other places, it's really easy to try to fill a bunch of blocks outside of the world where it will call `setBlockState` but not change the block, so it relies on both the state before call and the return value to cut down on invalid log entries.

I'm happy to change it if that's what you prefer, I just wanted to confirm.

Reply:
Yeah, please if you could, redirecting that logic seems like it'll cause trouble, and modifyArgs makes sure that everything used is what's actually being set

Reply:
So I had to replace Redirect with an Inject capturing local args rather than ModifyArgs because ModifyArgs doesn't support @Coerce which is required because FillCommand.Mode is private. I saw some of the other mixins were using injects with local captures so hopefully that's fine.